packages/compiler/src/service/parser/github-action/index.ts

```diff
@@ -0,0 +1,79 @@
+import fs from 'fs'
+import path from 'path'
+import decode from '../../../decode'
+import { ParamType, Service } from '../../type'
+import Parser from '../type'
+
+const relativePath = [
+  __dirname,
+  '..',
+  '..',
+  '..',
+  '..',
+  'src',
+  'service',
+  'parser',
+  'github-action',
+]
+
+export default class GithubAction implements Parser {
+  static fileName = 'action.yml'
+
+  match(dir: string): boolean {
+    return fs.existsSync(path.join(dir, GithubAction.fileName))
+  }
+
+  async parse(dir: string): Promise<Service> {
+    const filepath = path.join(dir, GithubAction.fileName)
+    if (!fs.existsSync(filepath)) throw new Error('invalid service')
+    const definition = decode(filepath, {})
+
+    if (fs.existsSync(path.join(dir, 'Dockerfile')))
+      throw new Error(
+        'Github action based on docker image are not supported yet',
+      )
+
+    fs.writeFileSync(
+      path.join(dir, 'Dockerfile'),
+      fs.readFileSync(path.join(...relativePath, 'Dockerfile')).toString(),
+    )
+
+    fs.writeFileSync(
+      path.join(dir, 'liteflow.js'),
+      fs
+        .readFileSync(path.join(...relativePath, 'liteflow.js'))
+        .toString()
+        .replace(
+          '__PATH__',
+          './' + ((definition.runs || {}).main || 'index.js'),
+        ),
+    )
+    return {
+      name: definition.name,
+      description: definition.description,
+      env: (definition.runs || {}).env || [],
+      events: [],
+      tasks: [
+        {
+          key: 'execute',
+          inputs: Object.keys(definition.inputs).map((key) => ({
+            key,
+            name: key,
+            description: definition.inputs[key].description,
+            type: ParamType.Any,
```

Comment:
All variable should be string:
```suggestion
            type: ParamType.String,
```